src/controllers/authControllers.ts

```diff
@@ -0,0 +1,35 @@
+import { Response, Request, NextFunction } from "express";
+import passport from "passport";
+
+type AuthHandler = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => Promise<void> | void;
+
+const loginController: AuthHandler = (req, res, next) => {
+  const { username, password } = req.body;
+
+  if (!username || !password) {
+    res.status(400).json({ message: "Username and password are required." });
+    return;
+  }
+
+  passport.authenticate(
+    "local",
+    { session: false },
+    (err: Error | null, user: any, info: { message: string }) => {
```

Comment:
Do we know the specific type for user? Is user coming for the database? We can just create a type or interface for this instead of using any. We might eventually have to create a type/interface library once we have our tables more ironed out.

Reply:
I have imported the User type from getPassword.ts, and replaced the `any` type here